### PR TITLE
Add nodeRef to Transition to avoid use of findDOMNode

### PR DESCRIPTION
**Pull request by ravicious**
teleport.e counterpart: https://github.com/gravitational/teleport.e/pull/5023

If `Transition` and `CSSTransition` are not provided the `nodeRef` prop, [they use the deprecated `findDOMNode` React API](https://reactcommunity.org/react-transition-group/transition#Transition-prop-nodeRef). This PR makes sure we do pass that prop so that we're ready for React 19.

It'd be nice to have this in before the test plan so that the change can be thoroughly tested.

### CODE REVIEW

File: web/packages/design/src/Dialog/Dialog.tsx [hunk 1]
```diff
@@ -16,42 +16,45 @@
...
+>((props, ref) => {
+  const { children, dialogCss, className, ...modalProps } = props;
+  return (
+    <Modal disableBackdropClick disableEscapeKeyDown {...modalProps}>
+      <ModalBox>
+        {/*
+            ref is supposed to be set on DialogBox, not Modal, because when used with
+            react-transition-group, it's DialogBox that's going to have its className set based on
+            the transition state.
+          */}
```
Comment (ravicious) [inline web/packages/design/src/Dialog/Dialog.tsx:45]: Passing a ref to `<Dialog>` was needed because in teleport.e, [`ReviewRequests` has a transition on `NotificationRoutingRulesDialog`](https://github.com/gravitational/teleport.e/pull/5023/files#diff-5e9a8e166ee07cdf201ef80f7268d94e3699a81410947bb443959de133abcce6) which then [renders a `Dialog`](https://github.com/gravitational/teleport.e/pull/5023/files#diff-a664fc40de9fd59cfa28fc83f4724d951b18cc4c781fad935878f501f4de2bed).

File: web/packages/design/src/Modal/index.ts [hunk 2]
```diff
@@ -18,3 +18,5 @@
 
 import Modal from './Modal';
 export default Modal;
+
+export type { Props } from './Modal';
```
Comment (kimlisa) [inline web/packages/design/src/Modal/index.ts:22]: fyi, after rebasing (b/c i ran into build issues), there was an error where there were duplicate Prop export

**Review by kimlisa — APPROVED** [review]
clicked around my staging, lgtm

File: web/packages/design/src/Dialog/Dialog.tsx [hunk 3]
```diff
@@ -16,42 +16,45 @@
...
+  PropsWithChildren<
+    {
+      className?: string;
+      dialogCss?: StyleFunction<any>;
+    } & ModalPropsWithoutChildren
+  >
+>((props, ref) => {
+  const { children, dialogCss, className, ...modalProps } = props;
+  return (
+    <Modal disableBackdropClick disableEscapeKeyDown {...modalProps}>
```
Comment (bl-nero) [inline web/packages/design/src/Dialog/Dialog.tsx:39]: Why are we changing these options? How is this related?

**Review by bl-nero — APPROVED** [review]

File: web/packages/design/src/Dialog/Dialog.tsx (hunk 3 above)
Comment (gzdunek) [reply web/packages/design/src/Dialog/Dialog.tsx:39]: We had them before as: 
```tsx
Dialog.defaultProps = {
   disableBackdropClick: true,
   disableEscapeKeyDown: true,
 };
```
So the behavior stays the same.

File: web/packages/design/src/Modal/index.ts (hunk 2 above)
Comment (gzdunek) [reply web/packages/design/src/Modal/index.ts]: Fixed.
